processing/src/main/java/org/apache/druid/segment/vector/VectorColumnSelectorFactory.java

```diff
@@ -57,6 +57,15 @@ default int getMaxVectorSize()
    */
   MultiValueDimensionVectorSelector makeMultiValueDimensionSelector(DimensionSpec dimensionSpec);
 
+  /**
+   * Returns an object selector for string columns, useful for non-dictionary encoded strings, or when
```

Comment:
What does it do when called on a non-string column? Cast the stuff to strings? Return null? Something else?

Reply:
hmm, i have no idea why I named this function like this, it should probably be called `makeDimensionObjectSelector`. Its just expected to make a `VectorObjectSelector` from a `DimensionSpec` rather than a raw column name

Reply:
The javadoc here should be fleshed out to describe the expected contract. IMO it should answer questions like:

- If you call this with a DimensionSpec that refers to a non-string column, what happens?
- Will the returned VectorObjectSelector always return a String or `null`?
- …might it return List<String> for a multi-value string column?

Reply:
I think javadocs are maybe cooler now in describing the intent of these methods